### PR TITLE
Issues with pdf application downloads

### PR DESCRIPTION
**Pull request by SamJamCul**
## Context
https://trello.com/c/AZSwZwcx/2568-issues-with-pdf-application-downloads

Applications can be downloaded as PDFs, but they were being formatted as American Letter pages, rather than A4 pages. Additionally, they took a relatively unmodified version of the page, which included a lot of 'webby' bits that weren't very high value, and took up page space unecessarily. 

## Changes proposed in this pull request
* Change application PDF downloads to A4 sizing
* Added no print flags to a selection of elements on the page, so that they don't show up in the PDF.

### Before
image

### After
image

## Guidance to review

You can check the PDF page sizing by downloading an application as a PDF using the link on the application page here 

image

If you're on a Mac, you can open the PDF from the finder, and from the top menu under Tools, use the inspector to view the page dimensions. A4 pages should be 210mm x 297mm. 

You can check the components that have been removed by downloading the PDF for an application as well, but if you want to inspect the code itself, the class `govuk-!-display-none-print` has been added to any elements that are deliberately being ignored

## Link to Trello card
https://trello.com/c/AZSwZwcx/2568-issues-with-pdf-application-downloads

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-teacher-training/blob/master/docs/environment-variables.md#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/views/layouts/_header.html.erb [hunk 1]
```diff
@@ -1,12 +1,12 @@
 <% case try(:current_namespace) %>
 <% when 'candidate_interface' %>
-  <%= render(HeaderComponent.new(classes: "app-header--#{HostingEnvironment.environment_name}",
+  <%= render(HeaderComponent.new(classes: "app-header--#{HostingEnvironment.environment_name} govuk-!-display-none-print",
```
Comment (duncanjbrown) [inline app/views/layouts/_header.html.erb:3]: I wonder if we should just add this to the `HeaderComponent`? It's now `govuk-!-display-none-print` everywhere it's used

File: app/views/layouts/_header.html.erb [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
                                  service_name: service_name,
                                  service_url: service_link,
                                  navigation_items: NavigationItems.for_candidate_interface(try(:current_candidate), controller))) %>
                                <%= render PhaseBannerComponent.new %>
 <% when 'support_interface' %>
-  <%= render(ProductHeaderComponent.new(classes: "app-header--#{HostingEnvironment.environment_name} app-header--full-border",
+  <%= render(ProductHeaderComponent.new(classes: "app-header--#{HostingEnvironment.environment_name} app-header--full-border govuk-!-display-none-print",
```
Comment (duncanjbrown) [inline app/views/layouts/_header.html.erb:9]: As above!

**Review by duncanjbrown — COMMENTED** [review]
Looking good! Would you consider also excluding

- the "SANDBOX FEATURE" block
- the expanding "View guidance given to Candidate" links

?

File: app/views/layouts/_header.html.erb (hunk 1 above)
Comment (SamJamCul) [reply app/views/layouts/_header.html.erb:3]: I did try to add it directly to the header component template, but it wasn't working. Was hoping that someone could help me understand why, seeing as it worked here when injected as the arg

Comment (SamJamCul) [reply app/views/layouts/_header.html.erb]: Sorted it out, turns out there's more than one type of header component, so I've flagged the correct ones.

**Comment by SamJamCul** [comment]
> 
> 
> Looking good! Would you consider also excluding
> 
>     * the "SANDBOX FEATURE" block
> 
>     * the expanding "View guidance given to Candidate" links
> 
> 
> ?

I've excluded these, and updated the preview image accordingly!

**Review by duncanjbrown — APPROVED** [review]
Thanks @SamJamCul!

I've just realised this won't include the offer any more as it's now on a separate tab. I'll ask URs whether they might want us to the offer back into the exported PDF.

**Pull request merged**